yarn-project/acir-simulator/src/avm/opcodes/instruction_impl.ts

```diff
@@ -0,0 +1,75 @@
+import { BufferCursor } from '../serialization/buffer_cursor.js';
+import {
+  Opcode,
+  OperandPair,
+  OperandType,
+  deserialize,
+  serialize,
+} from '../serialization/instruction_serialization.js';
+import { Instruction } from './instruction.js';
+
+export abstract class TwoOperandInstruction extends Instruction {
+  // Instruction wire format with opcode.
+  private static readonly wireFormat: OperandPair[] = [
+    [(c: TwoOperandInstruction) => c.opcode, OperandType.UINT8],
+    [(c: TwoOperandInstruction) => c.indirect, OperandType.UINT8],
+    [(c: TwoOperandInstruction) => c.inTag, OperandType.UINT8],
+    [(c: TwoOperandInstruction) => c.aOffset, OperandType.UINT32],
+    [(c: TwoOperandInstruction) => c.dstOffset, OperandType.UINT32],
+  ];
```

Comment:
Nice! So this is a base class providing serialization for the common case of two operand instructions with indirect and inTag? Very nice. Could you add a comment on each class explaining that?